### PR TITLE
Improve E0025, E0046, E0062, E0063 error explanations and add E0370 error explanation

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @Manishearth

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
@Manishearth: I'll stop this PR here. This file should be fine now.

File: src/librustc_typeck/diagnostics.rs [hunk 1]
```diff
@@ -615,14 +669,61 @@ variadic functions (except for its C-FFI).
...
+enum variant, one of the fields was specified more than once. Erroneous code
+example:
+
+```
+struct Foo {
+    x: i32
+}
+
+fn main() {
+    let x = Foo { x: 0,
```
Comment (Manishearth) [inline src/librustc_typeck/diagnostics.rs]: nit: Keep the fields on a new line

```
    let x = Foo {
                x: 0,
                x: 0
            };
```

File: src/librustc_typeck/diagnostics.rs [hunk 2]
```diff
@@ -1627,7 +1734,7 @@ It is not possible to declare type parameters on a function that has the `start`
 attribute. Such a function must have the following type signature:
 
 ```
-fn(isize, *const *const u8) -> isize
+fn(isize, *const *const u8) -> isize;
```
Comment (Manishearth) [inline src/librustc_typeck/diagnostics.rs]: Meh, semicolon not really necessary.

**Comment by Manishearth** [comment]
LGTM modulo nits

File: src/librustc_typeck/diagnostics.rs (hunk 2 above)
Comment (GuillaumeGomez) [inline src/librustc_typeck/diagnostics.rs]: I prefer sinc we're supposed to compile this code someday (yes, we are !).

**Comment by GuillaumeGomez** [comment]
@Manishearth: All good !

**Pull request merged**